### ISSUE TITLE
Attribute Error When (Re)Indexing a Model

### ISSUE THREAD

**Issue by johngriebel** [algolia/algoliasearch-django#226]
Hi,
This issue is similar to #13, but definitely distinct in my opinion. 

I've followed the [setup instructions](https://www.algolia.com/doc/api-client/django/setup/) for a model (`User`) in our [application](https://github.com/3Blades/app-backend). Here is the definition for my Index class:
```
class UserAlgoliaIndex(AlgoliaIndex):
    # fields = ('username', 'email', 'first_name', 'last_name')
    settings = {'searchableAttributes': ['username', 'email',
                                         'first_name', 'last_name']}
    index_name = 'users'
```

When I uncomment the `fields` line, things work as expected. However, if I remove that line in an attempt to get all fields from the `User` model, I get the following exception:
`AttributeError: 'User' object has no attribute 'logentry'`.

I'm sure you know this ultimately stems from this line:
`all_model_fields = [f.name for f in model._meta.get_fields()]`
(89 in models.py).

For the sake of completeness, here is the result of that line for my `User` model:
```['logentry', 'oauth2_provider_application', 'grant', 'accesstoken', 'refreshtoken', 'social_auth', 'userobjectpermission', 'notifications', 'notificationsettings', 'profile', 'emails', 'customer', 'projects', 'collaborator', 'projectfile', 'servers', 'actions', 'nodes', 'triggers', 'teams_created', 'groups_created', 'password', 'last_login', 'is_superuser', 'first_name', 'last_name', 'email', 'is_staff', 'is_active', 'date_joined', 'username', 'id', 'groups', 'user_permissions', 'team_groups']```.

Of course `logentry` (and many others) is not an actual field on `User`, but instead a foreign key that references it. I'm not suggesting, or asking, that these relationships be traversed with a call to `logentry_set.all()` (I'm not even sure what it would mean to do so), but instead to just skip them altogether. In other words, change 
`all_model_fields = [f.name for f in model._meta.get_fields()]` to 
`all_model_fields = [f.name for f in model._meta.get_fields() if not f.is_relation]`. 

I have tested this change, and it seems to work. I know this can be worked around by manually defining the model fields I would like returned, but I do feel that this would lower the barrier of entry for using Algoliasearch with Django.

Perhaps there is a good reason for not doing it this way currently, or maybe I missed something in the documentation. If you feel this would be useful, I can certainly submit a PR.

**Comment by PLNech** [comment]
Hi @johngriebel, thanks for your report and investigation! 

Indeed the problem is this `all_model_fields` initialization, and the solution you propose looks good to me and would simplify using Algolia with Django. We would be pleased to welcome your contribution to this project 🙂 

If you want to submit a PR, could you:
- [ ] Implement the fix you propose
- [ ] Check if the same can be done for `all_model_fields = model._meta.get_all_field_names()` (which will be the actual `all_model_fields` value on Django < 1.10)
- [ ] Add a test in `test_index.py` to confirm the resolution across all supported `Django x Python` versions

Once you open a PR with these, I'll be happy to review and merge it.

**Comment by johngriebel** [comment]
@PLNech Great! Hopefully I'll be able to get something together shortly.

**Comment by PLNech** [comment]
Closed by #227/#233. Thanks @johngriebel!
